Update plugin docs - devdot launch

laurapacilio opened the pull request:
## What

As we prepare content for migration to the new developer portal, we noticed that there are two separate sidebar entries for plugins in the Packer documentation. This situation is confusing for users, as it's not immediately obvious what documentation is where. The Packer plugin docs are also a big confusing because they use partials to display a large amount of the plugin install, usage, and "what are plugins" content in two separate places. This PR attempts to rectify this situation by making the following changes:

- **Creating one unified "Plugins" section in the sidebar.** Condenses plugin information into one place where users can find what they need. The Plugin overview page contains a clear link to the External Plugin documentation. Please note that we have another way to call out and link users directly to the External Plugin documentation from within the developer portal. Don't worry -- we'll make sure users can find it! :) 
- **Removing several plugins partials.** The bulk of plugin information (installation, what are plugins, etc.) has been revamped and is integrated into the Plugins sidebar section. The "External Plugins" documentation section now only contains relevant information about the types of External plugins and their badges. The overview page links back to the Packer Plugins main section in the documentation, where users can get more information about developing and installing these plugins, as needed. 
- **Creates a separate installation page**: Users may be specifically looking for installation information, and they may not expect to find it on a page called "Overview". 
- **Renames "Extending Packer" to "Developing Plugins":** First, this helps us keep parallelism with Terraform, which has a “Developing Plugins” section as well. Also, Extending Packer" doesn't help a reader figure out whether the section is relevant to them the way “Developing Plugins” does. A packer user surely knows whether they are up for the task of writing a plugin. But they may not think that "Extending Packer" implies "Developing Plugins".
- **Add a header to the External Plugin Docs sidebar**: It looks way better now! The sidebar before felt odd because it didn't have a header.
- **Moves HCP Support under Developing Plugins:** This page is relevant to plugin development and makes the most sense grouped with the other pages on this topic. I also updated all the page introductions to link back to HCP Support, where appropriate!

Some content has also been revised for clarity, concision, and style. For example, the installation instructions have a new intro, but remain largely the same. 

## Why

We believe these changes will help users better find the information they need within our documentation. The plugin confusion was also flagged as a blocker for the devdot launch.

## Screenshots

**New Sidebar**
<img width="980" alt="Screen Shot 2022-09-13 at 3 17 22 PM" src="https://user-images.githubusercontent.com/83350965/189990338-a47ce4ec-4515-444b-a6f3-8836b224093e.png">

**New Overview Page**
<img width="955" alt="Screen Shot 2022-09-13 at 3 16 20 PM" src="https://user-images.githubusercontent.com/83350965/189990185-860dea14-3a40-4984-a9ec-57b16c6f75fb.png">

**Revised Intro to Developing Plugins page**
<img width="1314" alt="Screen Shot 2022-09-12 at 5 32 21 PM" src="https://user-images.githubusercontent.com/83350965/189762439-2124e484-7635-4d78-bb09-f87b744e9d24.png">

**New External Plugins Sidebar + Overview Page**
<img width="906" alt="Screen Shot 2022-09-12 at 5 33 11 PM" src="https://user-images.githubusercontent.com/83350965/189762552-fb79e8ba-0249-4dfc-b42d-7b37aaef7ead.png">

nywilken commented on website/content/docs/plugins/index.mdx:
```diff
@@ -1,29 +1,21 @@
-
-@include "plugins/installing-plugins.mdx"
+Refer to the following plugin documentation:
+- [External Plugins](/plugins) to review the documentation for available packer plugins
```
nit: on the product name.
```suggestion
- [External Plugins](/plugins) to review the documentation for available Packer plugins
```

nywilken commented on website/content/docs/plugins/install-plugins.mdx:
```diff
@@ -1,17 +1,23 @@
-## Installing Plugins
+---
+description: |
+  Install external packer plugins that extend Packer functionality.
```
```suggestion
  Install external Packer plugins that extend Packer functionality.
```

nywilken commented on website/content/docs/plugins/index.mdx:
```diff
@@ -1,29 +1,21 @@
+- [External Plugins](/plugins) to review the documentation for available packer plugins
+- [Installing Plugins](/docs/plugins/install-plugins) to add external plugins to your Packer template and install the binaries
+- [Developing Plugins](/docs/plugins/creation) to get started creating custom external plugins
+- [HCP Support](/docs/plugins/hcp-support) to update a plugin to support HCP Packer
```
The HCP Packer support documentation explains the process for updating a custom builder to so that I can publish its artifacts to the HCP Packer registry. Should this sidebar be moved under the Developing Plugins section? 

Maybe even have the documentation linked to from the Customer Builders and Custom Post-Processors pages. What do you think?

laurapacilio replied:
I like it :) let me push up a few changes for you to look at!

nywilken reviewed (commented):
@laurapacilio thanks for taking on this work, and for the detailed PR description the changes look good to me. I left a couple of nits and a question about the consolidation of the HCP Packer support docs, as I think it should go under the Developing Plugins section.

laurapacilio commented:
okay @nywilken let me know what you think of these edits!
- I moved the HCP Support page to be underneath developing and edited all the page intros to link to it where appropriate (and to...you know, fix up the language a bit!)
- I also updated the overview page to clarify a bit more about built-in plugins AND link directly to those docs. This connects these concepts a bit better. We can do a follow-up PR where we rearrange the sidebar a bit more, but I want to avoid doing TOO MUCH in this one PR

nywilken reviewed (approved):
[Site Preview](https://packer-6vgeu7t22-hashicorp.vercel.app/docs) looks great. Thanks for the updated changes.

laurapacilio commented on website/data/docs-nav-data.json:
```diff
@@ -822,14 +818,22 @@
+      },
+      {
+        "title": "External Plugins",
+        "href": "/packer/plugins"
```
```suggestion
        "href": "/plugins"
```

laurapacilio commented:
Please note that the autogenerated backport PRs failed for some reason, so I manually cherry-picked the squashed and merged commit from this PR to both the `stable-website` and the `release/1.8.x` branch.
